test/lshr_shl_i1_regularize.ll

```diff
@@ -0,0 +1,40 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv -s %t.bc -o %t.reg.bc
+; RUN: llvm-dis %t.reg.bc -o - | FileCheck --check-prefix=CHECK-LLVM %s
+
+target triple = "spir64-unknown-unknown"
+
+@G.0 = addrspace(1) global i1 false
+@G.1 = addrspace(1) global i1 true
+
+define spir_func void @test_lshr_i1(i1 %a, i1 %b) {
+entry:
+  %0 = lshr i1 %a, %b
+; CHECK-LLVM: [[AI32_0:%[0-9]+]] = select i1 %a, i32 1, i32 0
+; CHECK-LLVM: [[BI32_0:%[0-9]+]] = select i1 %b, i32 1, i32 0
+; CHECK-LLVM: [[LSHRI32:%[0-9]+]] = lshr i32 [[AI32_0]], [[BI32_0]]
+; CHECK-LLVM: [[TRUNC_0:%[0-9]+]] = icmp ne i32 [[LSHRI32]], 0
+  %1 = zext i1 %0 to i32
```

Comment:
What is the purpose of these 'dead' instructions?

Reply:
I believe it's a result of the work of llvm fuzzer

Reply:
@asudarsa This is a check if we are preserving previous behavior. If you look on removed code you will see that `zext` instructions extending result of shift were removed because their presence was redundant.